internal/goldendataset/metric_gen.go

```diff
@@ -63,11 +68,25 @@ func DefaultCfg() MetricCfg {
 	}
 }
 
+// DefaultMetricData produces MetricData with a default config.
 func DefaultMetricData() data.MetricData {
 	return MetricDataFromCfg(DefaultCfg())
 }
 
+// MetricDataFromCfg produces MetricData with the passed-in config.
 func MetricDataFromCfg(cfg MetricCfg) data.MetricData {
+	return newMetricGenerator().genMetricDataFromCfg(cfg)
+}
+
+type metricGenerator struct {
+	metricID int
+}
+
+func newMetricGenerator() *metricGenerator {
+	return &metricGenerator{}
+}
```

Comment:
`metricID` is a counter that gets incremented every time a new metric is generated. It's used to create unique metric names. I guess we could hold and increment the current count on the stack, but it would be a more complicated solution that this.